lib/prelude/src/macros.rs

```diff
@@ -0,0 +1,54 @@
+//! This macro defines set of common macros which are useful across different projects.
+
+
+/// Allows for nicer definition of impls, similar to what Haskell or Scala does. Reduces the needed
+/// boilerplate. For example, the following usage:
```

Comment:
2 - IntelliJ can generate this code for me. And generated code is better for new code-readers.

Reply:
But intellij cannot generate code from docs, can it? So reading these sources you dont see immediately whats going on there

Reply:
Ah, I meant the macro itself, not the doc (i just comment docs, because there is a explanation of macro)

So I meant, that I don't see many benefits of using this macro, because the boilerplate IDE can generate anyway, and each macro extends the time one requires for being introduced to code.

Reply:
Oh, by usign this macro the impls are much shhorter and if I create a few similar ones I can easily make them line by line and understand them better! :)